Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1168,32 +1200,72 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
 {
 }
 
-void UnifiedPDFPlugin::setActiveAnnotation(RetainPtr<PDFAnnotation>&& annotation)
+void UnifiedPDFPlugin::setActivePDFPluginAnnotation(RetainPtr<PDFAnnotation>&& annotation)
 {
 #if PLATFORM(MAC)
     if (!supportsForms())
         return;
 
-    if (m_activeAnnotation)
-        m_activeAnnotation->commit();
+    if (m_activePDFPluginAnnotation)
+        m_activePDFPluginAnnotation->commit();
 
     if (annotation) {
 ALLOW_DEPRECATED_DECLARATIONS_BEGIN
         if ([annotation isKindOfClass:getPDFAnnotationTextWidgetClass()] && static_cast<PDFAnnotationTextWidget *>(annotation).isReadOnly) {
-            m_activeAnnotation = nullptr;
+            m_activePDFPluginAnnotation = nullptr;
             return;
         }
 ALLOW_DEPRECATED_DECLARATIONS_END
 
         auto activeAnnotation = PDFPluginAnnotation::create(annotation.get(), this);
-        m_activeAnnotation = activeAnnotation.get();
+        m_activePDFPluginAnnotation = activeAnnotation.get();
         activeAnnotation->attach(m_annotationContainer.get());
     } else
-        m_activeAnnotation = nullptr;
+        m_activePDFPluginAnnotation = nullptr;
     updateLayerHierarchy();
 #endif
 }
 
+void UnifiedPDFPlugin::startAnnotationTracking(RetainPtr<PDFAnnotation>& annotation)
+{
+    ASSERT(!m_trackedAnnotation);
+    m_trackedAnnotation = annotation;
+
+    if ([annotation isKindOfClass:getPDFAnnotationButtonWidgetClass()])
+        [annotation setHighlighted: YES];
+
+    updateLayerHierarchy();
+}
+
+void UnifiedPDFPlugin::finishAnnotationTracking()
+{
+    ASSERT(m_trackedAnnotation);
+
+    if ([m_trackedAnnotation isHighlighted])
+        [m_trackedAnnotation setHighlighted: NO];
+
+    if ([m_trackedAnnotation isKindOfClass:getPDFAnnotationButtonWidgetClass()] && [m_trackedAnnotation widgetControlType] != kPDFWidgetPushButtonControl) {
+        auto currentButtonState = [m_trackedAnnotation buttonWidgetState];
+        if (currentButtonState == PDFWidgetCellState::kPDFWidgetOnState && [m_trackedAnnotation allowsToggleToOff])
+            [m_trackedAnnotation setButtonWidgetState: PDFWidgetCellState::kPDFWidgetOffState];
+        else if (currentButtonState == PDFWidgetCellState::kPDFWidgetOffState)
+            [m_trackedAnnotation setButtonWidgetState: PDFWidgetCellState::kPDFWidgetOnState];
+    } else
+        ASSERT_NOT_IMPLEMENTED_YET();
```

Comment:
Won't this fire too much 😛

Reply:
It might! But perhaps it's ok in debug builds since this is an active construction site! Ok with removing if we think that's the better option instead

Reply:
🤷 eh, we'll see if it fires too much